### DISCUSSION TITLE
icon_cache leaks information

### DISCUSSION THREAD

**Discussion by mknj** [dani-garcia/vaultwarden#2338] [Ideas]
My Vaultwarden instance is used by some friends and i was surprised to find internal_server_name.miss entries of my friends employees in my icon_cache directory. There are also some other site entries that i really don't want to know about.

Would it be possible to disable the icon_cache completely or add an option to have an encrypted per user icon cache?

**Comment by RealOrangeOne** [comment]
Disabling outright would also remove another information leak, regarding a timing attack to work out whether there was a cache hit for a given domain or not.

You could set the icon cache to time very low, but that won't actually clean up the expired cache entries I believe.

**Comment by BlackDex** [comment]
@mknj i suggest to use icon redirect then instead of local cached.
See: https://github.com/dani-garcia/vaultwarden/blob/08f0de7b46d36a4e974d8e7b25a9786168cd38a5/.env.template#L132-L143

Encrypting the filename only will not work, because you can still open the file, or read metadata.
Just use the redirect option to prevent it from using the internal favicon downloader.
